File: app/src/main/java/ru/gsench/githubusers/domain/interactor/FavoriteObservable.java
package ru.gsench.githubusers.domain.interactor;

import java.io.IOException;
import java.util.ArrayList;

import ru.gsench.githubusers.domain.github_repo.ResponseParser;
import ru.gsench.githubusers.domain.utils.Pair;

/**
 * Created by grish on 12.05.2017.
 */

public class FavoriteObservable extends UserListObservable {

    private FavoritesManagement favorites;

    public FavoriteObservable(FavoritesManagement favorites) {
        super(FavoritesManagement.FAVORITES);
        this.favorites=favorites;
    }

    @Override
    public Pair<ArrayList<GitHubUserFavor>, Integer> obtain(int limit, int offset) throws IOException, ResponseParser.ParseException {
        return new Pair<>(favorites.getFavorites(), favorites.getFavorites().size());
    }
}

File: app/src/main/java/ru/gsench/githubusers/domain/interactor/FavoriteObservable.kt
package ru.gsench.githubusers.domain.interactor

import java.io.IOException
import java.util.ArrayList

import ru.gsench.githubusers.domain.github_repo.ResponseParser
import ru.gsench.githubusers.domain.utils.Pair

/**
 * Created by grish on 12.05.2017.
 */

class FavoriteObservable(private val favorites: FavoritesManagement) : UserListObservable(FavoritesManagement.FAVORITES) {

    @Throws(IOException::class, ResponseParser.ParseException::class)
    override fun obtain(limit: Int, offset: Int): Pair<ArrayList<UserModel>, Int> {
        return Pair(favorites.getFavorites(), favorites.getFavorites().size)
    }
}
